### PR TITLE
fix(Share): #MAG-301 fix share boards

### PR DESCRIPTION
**Pull request by Rseuret**
## Describe your changes
* rename file to keep old data
* fix shared option in toaster
* shared icon no longer displayed when shared array is empty

## Checklist tests
-Remove all shared right from a board must hide shared icon
-Old shared boards get the corrects rights
-Can share a board inside a folder 

## Issue ticket number and link
https://jira.support-ent.fr/browse/MAG-301

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/cgi/magneto/controller/ShareBoardController.java [hunk 1]
```diff
@@ -38,7 +41,7 @@ public class ShareController extends ControllerHelper {
     private final ServiceFactory serviceFactory;
     private final ShareBookMarkService shareBookMarkService;
 
-    public ShareController(ServiceFactory serviceFactory) {
+    public ShareBoardController(ServiceFactory serviceFactory) {
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/controller/ShareBoardController.java:44]: rename

File: src/main/java/fr/cgi/magneto/Magneto.java [hunk 2]
```diff
@@ -51,11 +51,11 @@ public void start() throws Exception {
 
         final EventBus eb = getEventBus(vertx);
 
-        ShareController shareController = new ShareController(serviceFactory);
-        addController(shareController);
-        shareController.setShareService(new MongoDbShareService(eb, MongoDb.getInstance(),
+        ShareBoardController shareBoardController = new ShareBoardController(serviceFactory);
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/Magneto.java:54]: rename

File: src/main/java/fr/cgi/magneto/Magneto.java [hunk 3]
```diff
@@ -51,11 +51,11 @@ public void start() throws Exception {
 
         final EventBus eb = getEventBus(vertx);
```
Comment (alicedraillard) [inline src/main/java/fr/cgi/magneto/Magneto.java:52]: nom du commit : fix share folder bugs

**Review by alicedraillard — APPROVED** [review]

**Review by alicedraillard — COMMENTED** [review]
nom des commits : mettre le format
fix([ce que tu fix]): #MAG-301 [description]
--> mettre les : après le fix, sans espace avant

**Review by alicedraillard — APPROVED** [review]

**Review by qperie — APPROVED** [review]

**Review by FlorentMr — APPROVED** [review]

**Pull request merged**